fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -547,10 +547,22 @@ private ScoredPlan planFilter(@Nonnull PlanContext planContext, @Nonnull QueryCo
         return withInJoin;
     }
 
-    @Nullable
-    private ScoredPlan planFilterWithInJoin(@Nonnull PlanContext planContext, @Nonnull InExtractor inExtractor, boolean needOrdering) {
-        int maxNumReplans = Math.max(getConfiguration().getMaxNumReplansForInToJoin(), 0);
-        boolean allowNonSargedInBindings = getConfiguration().getMaxNumReplansForInToJoin() < 0;
+    private static final class PlanWithInExtractor {
+        @Nonnull
+        private final ScoredPlan plan;
+        @Nonnull
+        private final InExtractor inExtractor;
+
+        PlanWithInExtractor(@Nonnull ScoredPlan plan, @Nonnull InExtractor inExtractor) {
+            this.plan = plan;
+            this.inExtractor = inExtractor;
+        }
+    }
+
+    private PlanWithInExtractor planFilterForInWithReplans(@Nonnull PlanContext planContext, @Nonnull InExtractor inExtractor, boolean needOrdering, int maxNumReplansInConfig) {
```

Comment:
Shouldn't that be `planFilterWithInWithReplans` :-) according to the existing naming scheme?